core/config/config.go

```diff
@@ -152,9 +152,16 @@ func Load() ConfigurationProvider {
 	}
 	baseCfg := NewProviderGroup("global", static...)
 
+	// TODO: (at) If application ID is not specified, the code lets the service initialization
```

Comment:
i think it should be a panic/hard failure if application ID is not specified. We need it for a lot places, not just config